org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/xsd/contentmodel/CMXSDElementDeclaration.java

```diff
@@ -309,6 +309,30 @@ private void collectElementsDeclaration(XSComplexTypeDefinition typeDefinition,
 		}
 	}
 
+	public HashMap<String, Boolean> isOptional(HashMap<String, Boolean> elementOptionality) {
+		XSTypeDefinition typeDefinition = elementDeclaration.getTypeDefinition();
+		switch (typeDefinition.getTypeCategory()) {
+			case XSTypeDefinition.SIMPLE_TYPE:
+				// TODO...
+				break;
+			case XSTypeDefinition.COMPLEX_TYPE:
+				XSParticle particle = ((XSComplexTypeDefinition) typeDefinition).getParticle();
+				if (particle != null) {
+					XSObjectList objectList = ((XSModelGroup) particle.getTerm()).getParticles();
+					for (int i = 0; i < objectList.getLength(); i++) {
+						XSParticle xp = (XSParticle) objectList.item(i);
+						XSTerm t = xp.getTerm();
+						if (t instanceof XSElementDeclaration) {
+							if ( xp != null ) {
+								elementOptionality.put(t.getName() ,xp.getMinOccurs() == 0);

```

Comment:
I return the information in a HashMap because I'm not sure if there is a way to access the element declaration of the parent from here?

Reply:
I suggest that you use an internal Map to have a clean API `boolean isOptional(String childElementName)`, see https://github.com/eclipse/lemminx/pull/1257/files#r928895910